compaction/compaction_manager.cc

```diff
@@ -669,6 +669,8 @@ compaction_manager::compaction_manager(config cfg, abort_source& as)
     , _strategy_control(std::make_unique<strategy_control>(*this))
     , _tombstone_gc_state(&_repair_history_maps)
 {
+    _compaction_module = make_shared<compaction_module>(tm);
+    tm.register_module(_compaction_module->get_name(), _compaction_module);
```

Comment:
we can move _compaction_module = make_shared<compaction_module>(tm) to constructor initializer list.

is the module automatically deregistered when destroyed?

Reply:
It's unregistered when stopped.
https://github.com/scylladb/scylladb/blob/96f62fd7a994acac606ab80ea50a180f7edfdcad/tasks/task_manager.cc#L252-L256
And module::stop() is called in compaction_manager::stop()